PWGCF/EbyEFluctuations/Tasks/MeanpTFlucPbPbIdentified.cxx

```diff
@@ -0,0 +1,432 @@
+      if (!((fabs(track.eta()) < 0.8) && (fabs(track.dcaXY()) < 0.12) && (fabs(track.dcaZ()) < 1.) && (track.pt() > 0.15 && track.pt() < 2.))) {
+        continue;
+      }
+
+      nCh += 1.;
+
+      Q1 += track.pt();
+      Q2 += (track.pt() * track.pt());
+
+      histos.fill(HIST("hP"), track.p());
+      histos.fill(HIST("hPt"), track.pt());
+      histos.fill(HIST("hEta"), track.eta());
+      histos.fill(HIST("hPtDCAxy"), track.pt(), track.dcaXY());
+      histos.fill(HIST("hPtDCAz"), track.pt(), track.dcaZ());
+
+      histos.fill(HIST("hPtEta"), track.pt(), track.eta());
+      histos.fill(HIST("hPEta"), track.p(), track.eta());
+
+      //    histos.fill(HIST("hPty"), track.pt(), track.rapidity());
+      //   histos.fill(HIST("hPy"), track.p(), track.rapidity());
+
+      VMeanPt.push_back(track.pt());
+
+      histos.fill(HIST("hNsigmaTPC"), track.p(), track.tpcNSigmaPr());
+
+      // only TPC tracks: Pion, Kaon, Proton
+      if (track.hasTPC() && abs(track.tpcNSigmaPi()) < 2.)
+        histos.fill(HIST("NSigamaTPCpion"), track.pt(), track.tpcNSigmaPi());
+      if (track.hasTPC() && abs(track.tpcNSigmaKa()) < 2.)
+        histos.fill(HIST("NSigamaTPCkaon"), track.pt(), track.tpcNSigmaKa());
+      if (track.hasTPC() && abs(track.tpcNSigmaPr()) < 2.)
+        histos.fill(HIST("NSigamaTPCproton"), track.pt(), track.tpcNSigmaPr());
+
+      // only TOF tracks: Pion, Kaon, Proton
+      if (track.hasTOF() && abs(track.tofNSigmaPi()) < 2.)
+        histos.fill(HIST("NSigamaTOFpion"), track.pt(), track.tofNSigmaPi());
+      if (track.hasTOF() && abs(track.tofNSigmaKa()) < 2.)
+        histos.fill(HIST("NSigamaTOFkaon"), track.pt(), track.tofNSigmaKa());
+      if (track.hasTOF() && abs(track.tofNSigmaPr()) < 2.)
+        histos.fill(HIST("NSigamaTOFproton"), track.pt(), track.tofNSigmaPr());
+
+      if (track.hasTPC())
+        histos.fill(HIST("hdEdx"), track.p(), track.tpcSignal());
+      if (track.hasTOF())
+        histos.fill(HIST("hTOFbeta"), track.p(), track.beta());
+
+      // only TPC+TOF tracks: Pion, Kaon, Proton
+      if ((track.hasTPC() && abs(track.tpcNSigmaPi()) < 2.) && (track.hasTOF() && abs(track.tofNSigmaPi()) < 2.))
+        histos.fill(HIST("NSigamaTPCTOFpion"), track.tpcNSigmaPi(), track.tofNSigmaPi());
+
+      if ((track.hasTPC() && abs(track.tpcNSigmaPi()) < 2. && (track.pt() > 0.2 && track.pt() < 0.5)))
+        histos.fill(HIST("hPtPion"), track.pt());
+
+      {
+        if ((track.hasTPC() && (track.pt() >= 0.5 && track.pt() < 2.0)) && (track.hasTOF() && abs(sqrt(track.tpcNSigmaPi()) * (track.tpcNSigmaPi()) + (track.tofNSigmaPi()) * (track.tofNSigmaPi())) < 2.)) {
+
+          histos.fill(HIST("hPtPion"), track.pt());
+          histos.fill(HIST("hEtaPion"), track.eta());
+
+          VMeanPtPion.push_back(track.pt());
+
+          for (ULong64_t jPi = 0; jPi < VMeanPtPion.size(); jPi++) {
+            histos.fill(HIST("hPtChPion"), nCh, VMeanPtPion[jPi]);
+          }
+
+          VMeanPtPion.clear();
```

Comment:
Consider you are doing this for every pion and you clear the vector after doing it
Is that what you wanted?

The same comment apply for the case of kaons and protons, I guess